functionary/core/utils/tasking.py

```diff
@@ -193,3 +206,47 @@ def _get_presigned_url(filename: str, environment: Environment) -> str:
     minio = MinioInterface(bucket_name=str(environment.id))
     presigned_url = minio.get_presigned_url(filename)
     return presigned_url
+
+
+def _start_function_task(task: Task) -> None:
+    """Publishes the task for execution"""
+    publish_task.delay(task.id)
+
+
+def _start_workflow_task(task: Task) -> None:
+    """Starts the workflow run associated with the given task"""
+    task.workflow.first_step.execute(workflow_task=task)
+
+
+def start_task(task: Task) -> None:
+    """Start the provided task
+
+    Starts a Task by executing the necessary steps for its tasked_object. The task
+    status will be set to PENDING and the task will be saved as part of this process.
+
+    Args:
+        task: Task to start
+
+    Returns:
+        None
+
+    Raises:
+        InvalidContentType: The tasked_object associated with the task is of an
+                            unrecognized type
+        InvalidStatus: The task cannot be started based on its current status
+    """
+    if task.status != Task.PENDING:
+        raise InvalidStatus(f"Task with status f{task.status} cannot be started")
+
+    task.in_progress()
+
+    tasked_type_class = task.tasked_type.model_class()
+
+    if tasked_type_class is Function:
```

Comment:
to pre-empt a potential comment here, match / case does not work when trying to match types.  At least I couldn't figure out how to get it to work and when I looked into how match / case works in general I'm not sure it's possible.